substrate/frame/glutton/src/lib.rs

```diff
@@ -208,6 +225,58 @@ pub mod pallet {
 		}
 	}
 
+	/// Dummy error type.
+	#[derive(codec::Encode, sp_runtime::RuntimeDebug)]
+	#[cfg_attr(feature = "std", derive(codec::Decode))]
+	pub enum InherentError {
+		/// Fatal error.
+		Fatal,
+	}
+
+	impl frame_support::inherent::IsFatalError for InherentError {
+		fn is_fatal_error(&self) -> bool {
+			matches!(self, InherentError::Fatal)
+		}
+	}
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = InherentError;
```

Comment:
```suggestion
		type Error = sp_inherents::MakeFatalError<()>;
```

The interface of `ProvideInherent` is currently more complicated than necessary.  
I think you can omit the `is_inherent_required` implementation, since `create_inherent` already returns `Some` in all cases.